src/Components/Aspire.Azure.AI.OpenAI/AspireConfigurableOpenAIExtensions.cs

```diff
@@ -0,0 +1,111 @@
+    /// <param name="connectionName">A name used to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    public static void AddOpenAIClientFromConfiguration(
+        this IHostApplicationBuilder builder,
+        string connectionName)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(connectionName);
+
+        var useAzure = false;
+
+        if (builder.Configuration.GetConnectionString(connectionName) is string connectionString)
+        {
+            useAzure = IsAzureConnectionString(connectionString, connectionName);
+        }
+
+        if (useAzure)
+        {
+            builder.AddAzureOpenAIClient(connectionName);
+        }
+        else
+        {
+            builder.AddOpenAIClient(connectionName);
+        }
+    }
+
+    /// <summary>
+    /// Registers <see cref="OpenAIClient"/> as a singleton in the services provided by the <paramref name="builder"/>.
+    /// The concrete implementation is selected automatically from configuration.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="name">The name of the component, which is used as the <see cref="ServiceDescriptor.ServiceKey"/> of the service and also to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    public static void AddKeyedOpenAIClientFromConfiguration(
+        this IHostApplicationBuilder builder,
+        string name)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        var useAzure = false;
+
+        if (builder.Configuration.GetConnectionString(name) is string connectionString)
+        {
+            useAzure = IsAzureConnectionString(connectionString, name);
+        }
+
+        if (useAzure)
+        {
+            builder.AddKeyedAzureOpenAIClient(name);
+        }
+        else
+        {
+            builder.AddKeyedOpenAIClient(name);
+        }
+    }
+
+    private static bool IsAzureConnectionString(string connectionString, string connectionName)
+    {
+        Uri? serviceUri = null;
+
+        var connectionBuilder = new DbConnectionStringBuilder
+        {
+            ConnectionString = connectionString
+        };
+
+        if (connectionBuilder.TryGetValue(ConnectionStringEndpoint, out var endpoint) && endpoint != null && Uri.TryCreate(endpoint.ToString(), UriKind.Absolute, out var endpointUri))
+        {
+            serviceUri = endpointUri;
+        }
+
+        if (serviceUri == null)
+        {
+            throw new InvalidOperationException($"An OpenAIClient could not be configured. Ensure valid connection information was provided in 'ConnectionStrings:{connectionName}' or specify a '{nameof(AzureOpenAISettings.Endpoint)}' or '{nameof(AzureOpenAISettings.Key)}' in the '{connectionName}' configuration section.");
```

Comment:
If there is no URI and only a Key, doesn't that mean `OpenAIClient` should be used?

Reply:
Correct, I fixed it after testing in eShop.